### PR TITLE
script: bump codespell to 2.2.1, update ignored words and fix spelling

### PR DESCRIPTION
**Pull request by jonatack**
as well as one in `test/lint/lint-locale-dependence.py` not seen by the spelling linter.

Can be tested locally by running `test/lint/lint-spelling.py` on this branch versus on master and by checking the CI linter result.

### CODE REVIEW

**Comment by aureleoules** [comment]
I opened #25706 recently to avoid PRs like these but it didn't get much attention. What are your thoughts about it?

**Comment by jonatack** [comment]
> I opened #25706 recently to avoid PRs like these but it didn't get much attention. What are your thoughts about it?

That pull wouldn't address all of the changes here; it wouldn't catch the one here in the linters, nor update the ignored words e.g. removing no-longer-needed ones as done here.

**Comment by brunoerg** [comment]
Could you please exemplify the changes in `test/lint/spelling.ignore-words.txt`? (e.g d575413fb8f8569803ed0050c15be003ef50951c added `desig` and now this PR removes it)

**Comment by jonatack** [comment]
> Could you please exemplify the changes in `test/lint/spelling.ignore-words.txt`? (e.g [d575413](https://github.com/bitcoin/bitcoin/commit/d575413fb8f8569803ed0050c15be003ef50951c) added `desig` and now this PR removes it)

Unsure what you are asking, but
- the code that you mention in your pull no longer uses `desig`
- the spelling linter passes without `desig`

**Comment by brunoerg** [comment]
> If the linter passes after removing any ignored words, we may as well remove them.
That was my doubt, but makes sense, thanks! In the future, if we need them again, we can put it back as well.

**Review by brunoerg — APPROVED** [review]
ACK b3b185ada59d11ee8a395554071da1d538e045a9

**Comment by MarcoFalke** [comment]
CI fails:

https://cirrus-ci.com/task/4515923582255104?logs=lint#L857

**Comment by jonatack** [comment]
> https://cirrus-ci.com/task/4515923582255104?logs=lint#L857

Thanks @MarcoFalke, bumped the codespell version.

**Review by satsie — APPROVED** [review]
ACK b6a65568dfbaf25839858b3114c28c07d8f9a45f

Was able to reproduce the CI errors with my local install of codespell. After bumping the version to 2.2.1, `test/lint/lint-spelling.py` ran with no errors.

Ran `test/lint/lint-spelling.py` on master and saw all the warnings that are fixed in this PR.

File: src/bench/nanobench.h [hunk 1]
```diff
@@ -858,7 +858,7 @@ class Bench {
      * @brief Retrieves all benchmark results collected by the bench object so far.
      *
      * Each call to run() generates a Result that is stored within the Bench instance. This is mostly for advanced users who want to
-     * see all the nitty gritty detials.
+     * see all the nitty gritty details.
```
Comment (MarcoFalke) [inline src/bench/nanobench.h:861]: this will need to be fixed upstream

**Pull request merged**